cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -33,11 +33,13 @@
 #include <rmm/cuda_stream_view.hpp>
 
 #include <cuda/std/optional>
+#include <cuda/std/type_traits>
 #include <thrust/iterator/counting_iterator.h>
 #include <thrust/iterator/transform_iterator.h>
 #include <thrust/pair.h>
 
 #include <algorithm>
+#include <type_traits>
```

Comment:
Why do we need two (same?) headers?

Reply:
I didn't replace host usage of traits like `std::is_same` in SFINAE. I only used the cuda::std traits in cases where we need device implementations to stop relying on constexpr (primarily for the `nullate` struct that needs to be passed to kernels etc).